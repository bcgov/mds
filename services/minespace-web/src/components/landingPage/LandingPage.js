import React from "react";
import { Divider, Button, Icon } from "antd";

import QuestionSidebar from "@/components/common/QuestionsSidebar";

import { USER_GUIDE, TAILINGS_DOCUMENT } from "@/constants/assets";

export const LandingPage = () => (
  <div>
    <div>
      <h1 className="landing-page-title"> Welcome to MineSpace, an online portal for BC mines</h1>
      <p>
        The Mines Digital Service (MDS) is a digital product being iteratively developed and
        designed to hold mine-related data that is user-friendly, trustworthy and reliable. The
        product will eventually replace the existing internal Mine Management System used by the
        Ministry and be used to surface information to the BC Mine Information website. The MDS
        project was started in July 2018 to help inspectors manage the mines they work with by
        replacing existing legacy systems and adding new functionality.
      </p>
      <br />
      <p>
        MineSpace is a piece of the Mines Digital Services Product that provides a link between the
        mine proponent and the Ministry. It is the online portal through which mines can interact
        with the Ministry by submitting reports and information as required by the Health, Safety,
        and Reclamation Code. MineSpace will enable mines to check the status of their reporting
        requirements and see the documentation that they have submitted to the Ministry for review.
        The portal is live and is expanding throughout 2019.
      </p>
    </div>
    <Divider className="gov-blue" style={{ height: "2px" }} />
    <div className="inline-flex block-tablet">
      <div className="flex-3">
        <h1 className="medium-margin-bot">Reporting templates</h1>
        <p>
          The Ministry is developing online templates to facilitate the submission of compliance
          information for Health and Safety, Geotechnical, Reclamation, and Geoscience. These
          templates will address Ministry requirements for Variance Applications, Incident
          Reporting, and the Tailings Storage Facility (TSF) Registry.
        </p>
        <br />
        <h2 className="medium-margin-bot">Tailings storage facilities and dams registry</h2>
        <p>
          Register of Tailings Storage Facilities and Dams (HSRC, 2016. 10.4.3). The registry should
          contain all tailings and water retaining structures on site, regardless of whether they
          are classified as dams or if they are currently operating or dormant structures.
        </p>
        <br />
        <p>
          <a href={TAILINGS_DOCUMENT}>
            <Icon type="download" /> Register of Tailings Storage Facilities and Dams (XLSM, 1.0 MB)
          </a>
          <br />
          <a href={USER_GUIDE}>
            <Icon type="download" /> User Guide - Register of Tailings Storage Facilities and Dams
            (PDF, 0.6 MB)
          </a>
        </p>
        <br />
        <p>
          <strong>
            Please submit completed registry spreadsheets to{" "}
            <a mailto="PERMRECL@gov.bc.ca">PERMRECL@gov.bc.ca</a>. You may also submit them via
            MineSpace if you participated in the 2019 Pilot Project.
          </strong>
        </p>

        <h1 className="large-margin-top medium-margin-bot">2019 pilot project</h1>
        <p>
          In March 2019, 6 mines were asked to participate in a pilot project to demonstrate how
          document submission would work in MineSpace. The pilot focused on the 2018 TSF annual
          reporting period and asked mines to submit required TSF reports through the MineSpace
          portal. During the pilot, an additional 26 mines were recruited by word of mouth and chose
          to participate. Following the pilot project, feedback from the participating mines showed
          that document submission through MineSpace was a success. It resulted in a total of 32
          mines voluntarily logging into MineSpace and submitting 120 required TSF reports. The
          submitted TSF reports were then instantly made available for Ministry staff to review -
          making it easier for both mines and Ministry staff to track document submission status.
        </p>
        <h1 className="large-margin-top medium-margin-bot">
          Prepare for next year&apos;s submission period
        </h1>
        <p>
          The pilot represents an initial step in mines interacting with Ministry through an online
          portal. We will require all mines to submit their required reporting through MineSpace
          starting in 2020, including variance requests, dangerous occurrence reports, and
          compliance information. To use this system you will need to get a business BCeID, and
          verify your existing credentials if you have not already already done so. This process can
          take time and it is best to begin as soon as possible.
        </p>
      </div>
      <div className="flex-1">
        <div className="sidebar-block">
          <div className="sidebar-block-title">
            <h2 className="side-bar-title">Do you have a Business BCeID?</h2>
          </div>
          <div className="sidebar-block-content">
            <p>
              In order to submit information through MineSpace, you will need to register for a
              Business BCeID. This process can take several weeks, so ensure you allow plenty of
              time before end of year reporting deadlines.
            </p>
            <br />
            <p>
              You will need to add employees and delegates who will be using the system to your
              Business BCeID account.
            </p>
            <div className="sidebar-button">
              <a
                href="https://www.bceid.ca/register/business/getting_started/getting_started.aspx"
                target="_blank"
                rel="noopener noreferrer"
              >
                <Button type="primary" size="large" block>
                  <span>Get a BCeID</span>
                </Button>
              </a>
            </div>
          </div>
        </div>
        <QuestionSidebar />
      </div>
    </div>
  </div>
);

export default LandingPage;
