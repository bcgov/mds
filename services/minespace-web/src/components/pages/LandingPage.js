import React from "react";
import { connect } from "react-redux";
import { Row, Col, Card, Button, Typography } from "antd";
import * as Strings from "@common/constants/strings";
import PropTypes from "prop-types";
import * as COMMON_ENV from "@common/constants/environment";
// Uncomment when image is re-introduced
// import { MAP_LOGO } from "@/constants/assets";
import * as MINESPACE_ENV from "@/constants/environment";
import { isAuthenticated } from "@/selectors/authenticationSelectors";

const { Paragraph, Text, Title } = Typography;

const propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
};

export const LandingPage = (props) => (
  <div>
    {/* Use this instead and not the below Row when we want to display the image! */}
    {/* <Row
      type="flex"
      justify="center"
      align="middle"
      gutter={[{ sm: 0, xl: 64 }]}
      className="landing-header"
    >
      <Col sm={1} xl={2} xxl={4} />
      <Col span={6}>
        <img src={MAP_LOGO} />
      </Col>
      <Col sm={22} xl={10}>
        <Title>Welcome to MineSpace</Title>
        <Paragraph className="header-text">
          Manage applications, see inspection histories, submit reports, and more.
        </Paragraph>
        <Button type="primary" size="large" className="login">
          <a
            href={`${COMMON_ENV.KEYCLOAK.loginURL}${MINESPACE_ENV.BCEID_LOGIN_REDIRECT_URI}&kc_idp_hint=${COMMON_ENV.KEYCLOAK.idpHint}`}
          >
            Log in
          </a>
        </Button>
      </Col>
      <Col sm={1} xl={2} xxl={4} />
    </Row> */}
    <Row
      type="flex"
      justify="center"
      align="top"
      className="landing-header"
      gutter={[{ sm: 0, xl: 64 }]}
    >
      <Col xl={{ span: 24 }} xxl={{ span: 20 }}>
        <Title>Welcome to MineSpace</Title>
        <Paragraph className="header-text">
          Manage applications, see inspection histories, submit reports, and more.
        </Paragraph>
        {!props.isAuthenticated && (
          <Button type="primary" size="large" className="login">
            <a
              href={`${COMMON_ENV.KEYCLOAK.loginURL}${MINESPACE_ENV.BCEID_LOGIN_REDIRECT_URI}&kc_idp_hint=${COMMON_ENV.KEYCLOAK.idpHint}`}
            >
              Log in
            </a>
          </Button>
        )}
      </Col>
    </Row>
    <Row
      gutter={[{ sm: 0, xl: 64 }]}
      type="flex"
      justify="center"
      align="top"
      className="landing-section"
    >
      <Col sm={{ span: 24 }} xl={{ span: 12 }} xxl={{ span: 10 }}>
        <Title level={4}>What is MineSpace?</Title>
        <Paragraph>
          The <Text strong>Ministry of Energy, Mines and Petroleum Resources</Text> is developing a
          system to make it easier for the public, industry and government to see what&apos;s
          happening in the mining industry across British Columbia. The system is called&nbsp;
          <Text strong>Mines Digital Services (MDS)</Text>.
        </Paragraph>
        <Paragraph>
          <Text strong>MineSpace</Text> is part of the MDS system, developed specifically for
          industry. It is intended to make it easier for businesses to manage applications, see
          their inspection history and submit reports.
        </Paragraph>
        <Paragraph>
          This system is being developed iteratively and with input from people who operate mines
          across B.C.
        </Paragraph>

        <Title level={4}>What can I do in MineSpace?</Title>
        <ul className="landing-list">
          <li>Upload any report specified in the Health, Safety and Reclamation Code</li>
          <li>View all code variances granted to and incidents reported by your mine</li>
          <li>View your mine permit and amendment history</li>
          <li>See all the contacts the Ministry has on file for your mine</li>
          <li>Find important Ministry contacts</li>
        </ul>
        <Paragraph />
      </Col>
      <Col sm={{ span: 24 }} xl={{ span: 12 }} xxl={{ span: 10 }}>
        <Title level={4}>How do I get access?</Title>
        <Paragraph>
          You must have a <Text strong>Business or Personal BCeID</Text> and then contact us to
          request access to MineSpace.
        </Paragraph>
        <Paragraph strong>If you have a BCeID:</Paragraph>
        <Paragraph>
          Contact us at&nbsp;
          <a href={`mailto:${Strings.MDS_EMAIL}`}>{Strings.MDS_EMAIL}</a>
          &nbsp;to request access to MineSpace.
        </Paragraph>
        <Paragraph strong>If you have multiple employees who need to use MineSpace:</Paragraph>

        <Paragraph>
          <Text>Add them to your Business BCeID</Text>
          <Text>Let us know you want them to be able to access MineSpace</Text>
        </Paragraph>

        <Title level={4}>Don&apos;t have a BCeID?</Title>
        <Paragraph>
          In order to access MineSpace, you need to register for a Business or Personal BCeID. It
          can take several weeks to process the request, so give yourself plenty of lead time.
        </Paragraph>
        <Paragraph>
          Once you have your BCeID, you can add employees and delegates. You can request that anyone
          added to your Business BCeID account be given access to MineSpace.
        </Paragraph>
        <Row type="flex" justify="center">
          <Col>
            <a
              href="https://www.bceid.ca/register/business/getting_started/getting_started.aspx"
              target="_blank"
              rel="noopener noreferrer"
            >
              <Button type="primary" size="large">
                Get a BCeID
              </Button>
            </a>
          </Col>
        </Row>
      </Col>
    </Row>
    <Row
      gutter={[{ sm: 0, xl: 64 }]}
      type="flex"
      justify="center"
      align="top"
      className="landing-section"
    >
      <Col sm={{ span: 24 }} xl={{ span: 12 }} xxl={{ span: 10 }}>
        <Card title="Questions?">
          <Row>
            <Col>
              <Paragraph>
                Please let us know about any questions or comments you have regarding your
                experience using MineSpace.
              </Paragraph>
              <Paragraph>
                Email us at&nbsp;
                <a href={`mailto:${Strings.MDS_EMAIL}`}>{Strings.MDS_EMAIL}</a>.
              </Paragraph>
            </Col>
          </Row>
        </Card>
      </Col>
    </Row>
  </div>
);

const mapStateToProps = (state) => ({
  isAuthenticated: isAuthenticated(state),
});

LandingPage.propTypes = propTypes;

export default connect(mapStateToProps)(LandingPage);
