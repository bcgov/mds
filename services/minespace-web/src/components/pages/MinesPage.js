import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Row, Col, Divider, Typography, Icon } from "antd";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { getUserInfo, isProponent } from "@/selectors/authenticationSelectors";
import { getUserMineInfo } from "@/selectors/userMineSelectors";
import { fetchUserMineInfo } from "@/actionCreators/userDashboardActionCreator";
import CustomPropTypes from "@/customPropTypes";
import Loading from "@/components/common/Loading";
import * as routes from "@/constants/routes";
import * as Strings from "@/constants/strings";
import Map from "@/components/common/Map";
import UnauthenticatedNotice from "../common/UnauthenticatedNotice";
import { detectDevelopmentEnvironment } from "@/utils/environmentUtils";

const { Paragraph, Title, Text } = Typography;

const propTypes = {
  userInfo: PropTypes.objectOf(PropTypes.string).isRequired,
  fetchUserMineInfo: PropTypes.func.isRequired,
  userMineInfo: CustomPropTypes.userMines.isRequired,
};

export class MinesPage extends Component {
  state = { isLoaded: false };

  componentDidMount() {
    this.props.fetchUserMineInfo().then(() => this.setState({ isLoaded: true }));
  }

  render() {
    const { mines } = this.props.userMineInfo;
    const isProdOrTest = !detectDevelopmentEnvironment();
    if (!this.props.isProponent && isProdOrTest) {
      return <UnauthenticatedNotice />;
    } else {
      return (
        (this.state.isLoaded && (
          <Row>
            <Col>
              <Title>My Mines</Title>
              <Divider />
              <Title level={4}>Welcome, {this.props.userInfo.preferred_username}.</Title>
              {(mines && mines.length > 0 && (
                <Row gutter={32}>
                  <Col xl={12} lg={14} sm={24}>
                    <Paragraph>
                      You are authorized to submit information for the following mines:
                    </Paragraph>
                    <div className="link-card">
                      <ul>
                        {mines
                          .sort((a, b) => (a.mine_name > b.mine_name ? 1 : -1))
                          .map((mine) => (
                            <li key={mine.mine_guid}>
                              <Link to={routes.MINE_DASHBOARD.dynamicRoute(mine.mine_guid)}>
                                <Title level={4}>
                                  <Icon type="environment" style={{ paddingRight: "5px" }} />
                                  {mine.mine_name}{" "}
                                </Title>
                                <Text>Mine Number: {mine.mine_no || Strings.EMPTY_FIELD}</Text>
                              </Link>
                            </li>
                          ))}
                      </ul>
                    </div>
                  </Col>
                  <Col xl={12} lg={10} sm={0}>
                    <div style={{ height: "400px", marginTop: "-32px" }}>
                      <Map
                        controls={false}
                        additionalPins={mines.map((mine) => [mine.latitude, mine.longitude])}
                      />
                    </div>
                    <Paragraph style={{ paddingTop: "16px" }}>
                      Don&apos;t see the mine you&apos;re looking for? Contact&nbsp;
                      <a href={`mailto:${Strings.MDS_EMAIL}`}>{Strings.MDS_EMAIL}</a>
                      &nbsp;for assistance.
                    </Paragraph>
                  </Col>
                </Row>
              )) || (
                <Row>
                  <Col>
                    <Paragraph>
                      You are not authorized to manage information for any mines. Please
                      contact&nbsp;
                      <a className="underline" href={Strings.MDS_EMAIL}>
                        {Strings.MDS_EMAIL}
                      </a>
                      &nbsp;for assistance.
                    </Paragraph>
                  </Col>
                </Row>
              )}
            </Col>
          </Row>
        )) || <Loading />
      );
    }
  }
}

const mapStateToProps = (state) => ({
  userInfo: getUserInfo(state),
  userMineInfo: getUserMineInfo(state),
  isProponent: isProponent(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchUserMineInfo,
    },
    dispatch
  );

MinesPage.propTypes = propTypes;

export default connect(mapStateToProps, mapDispatchToProps)(MinesPage);
