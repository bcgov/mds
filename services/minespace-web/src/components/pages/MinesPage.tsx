import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import EnvironmentOutlined from "@ant-design/icons/EnvironmentOutlined";
import { Row, Col, Divider, Typography, Button, Skeleton } from "antd";
import { Link } from "react-router-dom";
import { getUserInfo, isProponent } from "@mds/common/redux/selectors/authenticationSelectors";
import { getUserMinePageData } from "@/selectors/userMineSelectors";
import { fetchUserMineInfo } from "@/actionCreators/userDashboardActionCreator";
import * as routes from "@/constants/routes";
import * as Strings from "@/constants/strings";
import Map from "@/components/common/Map";
import UnauthenticatedNotice from "../common/UnauthenticatedNotice";
import { IMine, detectDevelopmentEnvironment } from "@mds/common";

const DEFAULT_MINES_PER_PAGE = 10;
const DEFAULT_MINE_PAGE = 1;

export const MinesPage = () => {
  const dispatch = useDispatch();
  const minePageData = useSelector(getUserMinePageData);
  const userIsProponent = useSelector(isProponent);
  const userInfo = useSelector(getUserInfo);

  const [isLoaded, setIsLoaded] = useState(false);
  const [desiredPage, setDesiredPage] = useState(DEFAULT_MINE_PAGE);

  const per_page = DEFAULT_MINES_PER_PAGE;
  const { mines, current_page, total, total_pages } = minePageData;
  const isProdOrTest = !detectDevelopmentEnvironment();
  const isUnauthenticated = !userIsProponent && isProdOrTest;

  useEffect(() => {
    // for "Can't perform a React state update on an unmounted component."
    let isMounted = true;
    setIsLoaded(false);
    const queryParams = `?page=${desiredPage}&per_page=${per_page}`;
    dispatch(fetchUserMineInfo(queryParams)).then(() => {
      if (isMounted) {
        setIsLoaded(true);
      }
    });

    return () => {
      isMounted = false;
    };
  }, [desiredPage]);

  return isUnauthenticated ? (
    <UnauthenticatedNotice />
  ) : (
    <Row>
      <Col span={24}>
        <Typography.Title>My Mines</Typography.Title>
        <Divider />

        <Typography.Title level={4}>Welcome, {userInfo.preferred_username}.</Typography.Title>

        <Skeleton loading={!isLoaded}>
          {(mines && mines.length > 0 && (
            <Row gutter={32}>
              <Col xl={12} lg={14} sm={24}>
                <Typography.Paragraph>
                  You are authorized to submit information for the following mines:
                </Typography.Paragraph>
                <div className="link-card">
                  <ul>
                    {mines
                      .sort((a, b) => (a.mine_name > b.mine_name ? 1 : -1))
                      .map((mine: IMine) => (
                        <li key={mine.mine_guid}>
                          <Link to={routes.MINE_DASHBOARD.dynamicRoute(mine.mine_guid)}>
                            <Typography.Title level={4}>
                              <EnvironmentOutlined style={{ paddingRight: "5px" }} />
                              {mine.mine_name}
                            </Typography.Title>
                            <Typography.Text>
                              Mine Number: {mine.mine_no || Strings.EMPTY_FIELD}
                            </Typography.Text>
                          </Link>
                        </li>
                      ))}
                  </ul>
                </div>
              </Col>
              <Col xl={12} lg={10} sm={0}>
                <div style={{ height: "400px", marginTop: "-32px" }}>
                  <Map
                    controls={false}
                    additionalPins={mines.map((mine) => [mine.latitude, mine.longitude])}
                  />
                </div>
                <Typography.Paragraph style={{ paddingTop: "16px" }}>
                  Don&apos;t see the mine you&apos;re looking for? Contact&nbsp;
                  <a href={`mailto:${Strings.MDS_EMAIL}`}>{Strings.MDS_EMAIL}</a>
                  &nbsp;for assistance.
                </Typography.Paragraph>
              </Col>
            </Row>
          )) || (
            <Row>
              <Col span={24}>
                <Typography.Paragraph>
                  You are not authorized to manage information for any mines. Please contact&nbsp;
                  <a className="underline" href={Strings.MDS_EMAIL}>
                    {Strings.MDS_EMAIL}
                  </a>
                  &nbsp;for assistance.
                </Typography.Paragraph>
              </Col>
            </Row>
          )}
        </Skeleton>
        {true && (
          <>
            <Row justify="space-between">
              <Button
                onClick={() => setDesiredPage(current_page - 1)}
                disabled={desiredPage <= DEFAULT_MINE_PAGE}
                loading={!isLoaded}
              >
                Previous
              </Button>
              <Button
                disabled={desiredPage >= total_pages}
                onClick={() => setDesiredPage(current_page + 1)}
                loading={!isLoaded}
              >
                Next
              </Button>
            </Row>
            Viewing page {current_page} of {total_pages} ({total} mines)
          </>
        )}
      </Col>
    </Row>
  );
};

export default MinesPage;
