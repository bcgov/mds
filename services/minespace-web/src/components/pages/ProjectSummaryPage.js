/* eslint-disable */
import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Link } from "react-router-dom";
import { getFormValues } from "redux-form";
import { Row, Col, Typography, Tabs, Divider } from "antd";
import { CaretLeftOutlined } from "@ant-design/icons";
import PropTypes from "prop-types";
import { getMines } from "@common/selectors/mineSelectors";
import {
  getProjectSummary,
  getFormattedProjectSummary,
} from "@common/selectors/projectSummarySelectors";
import {
  getProjectSummaryDocumentTypesHash,
  getProjectSummaryAuthorizationTypesArray,
} from "@common/selectors/staticContentSelectors";
import {
  createProjectSummary,
  fetchProjectSummaryById,
  updateProjectSummary,
} from "@common/actionCreators/projectSummaryActionCreator";
import { fetchMineRecordById } from "@common/actionCreators/mineActionCreator";
import { formatTitleString } from "@common/utils/helpers";
import * as FORM from "@/constants/forms";
import Loading from "@/components/common/Loading";
import { EDIT_PROJECT_SUMMARY, MINE_DASHBOARD, ADD_PROJECT_SUMMARY } from "@/constants/routes";
import CustomPropTypes from "@/customPropTypes";
import ProjectSummaryForm from "@/components/Forms/projectSummaries/ProjectSummaryForm";

const propTypes = {
  formValues: PropTypes.shape({
    project_summary_date: PropTypes.string,
    project_summary_description: PropTypes.string,
    documents: PropTypes.arrayOf(PropTypes.object),
  }),
  projectSummary: CustomPropTypes.projectSummary,
  fetchProjectSummaryById: PropTypes.func.isRequired,
  createProjectSummary: PropTypes.func.isRequired,
  updateProjectSummary: PropTypes.func.isRequired,
  fetchMineRecordById: PropTypes.func.isRequired,
  projectSummaryDocumentTypesHash: PropTypes.objectOf(PropTypes.string).isRequired,
  match: PropTypes.shape({
    params: {
      mineGuid: PropTypes.string,
    },
  }).isRequired,
  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
};

const defaultProps = {
  projectSummary: {},
  formValues: {},
};

const tabs = [
  "basic-information",
  "project-contacts",
  "project-dates",
  "authorizations-involved",
  "document-upload",
];
export class ProjectSummaryPage extends Component {
  state = {
    isLoaded: false,
    isEditMode: false,
    activeTab: tabs[0],
  };

  componentDidMount() {
    const { mineGuid, projectSummaryGuid, tab } = this.props.match?.params;
    this.props.fetchMineRecordById(mineGuid).then(() => {
      if (mineGuid && projectSummaryGuid) {
        return this.props.fetchProjectSummaryById(mineGuid, projectSummaryGuid).then(() => {
          this.setState({ isLoaded: true, isEditMode: true, activeTab: tab });
        });
      }
      return this.setState({ isLoaded: true, activeTab: tab });
    });
  }

  handleSubmit = (values) => {
    if (!this.state.isEditMode) {
      return this.handleCreateProjectSummary(values);
    }
    return this.handleUpdateProjectSummary(values);
  };

  handleTransformPayload = (values) => {
    const authorizations = [];
    Object.keys(values).map((key) => {
      if (this.props.projectSummaryAuthorizationTypesArray.includes(key)) {
        authorizations.push({
          project_summary_authorization_type: key,
          existing_permits_authorizations: values[key].existing_permits_authorizations.split(","),
          ...values[key],
        });
        delete values[key];
      }
    });

    return { authorizations, ...values };
  };

  handleCreateProjectSummary(values) {
    return this.props
      .createProjectSummary(
        {
          mineGuid: this.props.match.params?.mineGuid,
        },
        this.handleTransformPayload(values)
      )
      .then(({ data: { mine_guid, project_summary_guid } }) => {
        this.props.history.push(EDIT_PROJECT_SUMMARY.dynamicRoute(mine_guid, project_summary_guid));
        window.location.reload();
      });
  }

  handleTabChange = (activeTab) => {
    const url = this.state.isEditMode
      ? EDIT_PROJECT_SUMMARY.dynamicRoute(
          this.props.match.params?.mineGuid,
          this.props.match.params?.projectSummaryGuid,
          activeTab
        )
      : ADD_PROJECT_SUMMARY.dynamicRoute(this.props.match.params?.mineGuid, activeTab);
    this.setState({ activeTab });
    this.props.history.push(url);
  };

  handleUpdateProjectSummary(values) {
    const { mine_guid: mineGuid, project_summary_guid: projectSummaryGuid } = values;
    return this.props
      .updateProjectSummary(
        {
          mineGuid,
          projectSummaryGuid,
        },
        this.handleTransformPayload(values)
      )
      .then(({ data: { mine_guid, project_summary_guid } }) => {
        this.props.fetchProjectSummaryById(mine_guid, project_summary_guid);
      });
  }

  render() {
    console.log(this.props.projectSummary);
    console.log(this.props.formattedProjectSummary);
    const { mineGuid } = this.props.match?.params;
    const mine = this.props.mines[mineGuid];
    const title = this.state.isEditMode
      ? `Edit project description - ${this.props.projectSummary?.project_summary_title}`
      : `New project description for ${mine.mine_name}`;
    return (
      (this.state.isLoaded && (
        <>
          <Row>
            <Col span={24}>
              <Typography.Title>
                <Link to={MINE_DASHBOARD.dynamicRoute(mineGuid, "applications")}>
                  <CaretLeftOutlined />
                </Link>
                {title}
              </Typography.Title>
            </Col>
          </Row>
          <Divider />
          <Tabs
            tabPosition="left"
            activeKey={this.state.activeTab}
            defaultActiveKey={tabs[0]}
            onChange={this.handleTabChange}
          >
            {tabs.map((tab) => (
              <Tabs.TabPane tab={formatTitleString(tab)} key={tab}>
                <ProjectSummaryForm
                  initialValues={
                    this.state.isEditMode
                      ? this.props.formattedProjectSummary.summary
                      : {
                          contacts: [{ is_primary: true }],
                          status_code: "D",
                        }
                  }
                  mineGuid={mineGuid}
                  isEditMode={this.state.isEditMode}
                  onSubmit={this.handleSubmit}
                  projectSummaryDocumentTypesHash={this.props.projectSummaryDocumentTypesHash}
                  handleTabChange={this.handleTabChange}
                />
              </Tabs.TabPane>
            ))}
          </Tabs>
        </>
      )) || <Loading />
    );
  }
}

const mapStateToProps = (state) => ({
  mines: getMines(state),
  formValues: getFormValues(FORM.ADD_EDIT_PROJECT_SUMMARY)(state) || {},
  projectSummary: getProjectSummary(state),
  formattedProjectSummary: getFormattedProjectSummary(state),
  projectSummaryDocumentTypesHash: getProjectSummaryDocumentTypesHash(state),
  projectSummaryAuthorizationTypesArray: getProjectSummaryAuthorizationTypesArray(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      createProjectSummary,
      fetchProjectSummaryById,
      updateProjectSummary,
      fetchMineRecordById,
    },
    dispatch
  );

ProjectSummaryPage.propTypes = propTypes;
ProjectSummaryPage.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(ProjectSummaryPage);
