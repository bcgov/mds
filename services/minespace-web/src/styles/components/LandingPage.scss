.landing-header {
  padding-top: 64px;
  padding-bottom: 64px;

  .login {
    a {
      text-decoration: none;
    }
  }
}

.header-text {
  font-size: 24px;
}

.landing-section {
  padding-bottom: 32px;
  ul {
    padding: 0;
    margin: 0;
  }
  .ant-card {
    margin-top: 50px;
    margin-bottom: 22px;
  }
}
