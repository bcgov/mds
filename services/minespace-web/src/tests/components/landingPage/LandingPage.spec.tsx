import React from "react";
import { shallow } from "enzyme";
import { LandingPage } from "@/components/pages/LandingPage";

const props = { isAuthenticated: false };
describe("LandingPage", () => {
  it("renders properly", () => {
    const component = shallow(<LandingPage {...props} />);
    expect(component).toMatchSnapshot();
  });
});
