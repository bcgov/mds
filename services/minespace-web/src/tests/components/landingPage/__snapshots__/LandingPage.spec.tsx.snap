// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders properly 1`] = `
<div>
  <Row
    align="top"
    className="landing-header"
    gutter={64}
    justify="center"
  >
    <Col
      xl={
        Object {
          "span": 24,
        }
      }
      xxl={
        Object {
          "span": 20,
        }
      }
    >
      <ForwardRef>
        Welcome to MineSpace
      </ForwardRef>
      <ForwardRef
        className="header-text"
      >
        Manage applications, see inspection histories, submit reports, and more.
      </ForwardRef>
      <div>
        <LoginButton
          className="login"
        />
        <AuthorizationWrapper
          inTesting={true}
        >
          <Button
            className="login"
            size="large"
            type="primary"
          >
            <a
              href="<URL>http://localhostundefined/return-page?type=login&kc_idp_hint=ms-verifiable-credential"
            >
              Log in with Verifiable Credentials
            </a>
          </Button>
        </AuthorizationWrapper>
      </div>
    </Col>
  </Row>
  <Row
    align="top"
    className="landing-section"
    gutter={64}
    justify="center"
  >
    <Col
      sm={
        Object {
          "span": 24,
        }
      }
      xl={
        Object {
          "span": 12,
        }
      }
      xxl={
        Object {
          "span": 10,
        }
      }
    >
      <ForwardRef
        level={4}
      >
        What is MineSpace?
      </ForwardRef>
      <ForwardRef>
        The
         
        <ForwardRef(Text)
          strong={true}
        >
          Ministry of Energy, Mines and Low Carbon Innovation
        </ForwardRef(Text)>
         
        is developing a system to make it easier for the public, industry and government to see what's happening in the mining industry across British Columbia. The system is called 
        <ForwardRef(Text)
          strong={true}
        >
          Mines Digital Services (MDS)
        </ForwardRef(Text)>
        .
      </ForwardRef>
      <ForwardRef>
        <ForwardRef(Text)
          strong={true}
        >
          MineSpace
        </ForwardRef(Text)>
         is part of the MDS system, developed specifically for industry. It is intended to make it easier for businesses to manage applications, see their inspection history and submit reports.
      </ForwardRef>
      <ForwardRef>
        This system is being developed iteratively and with input from people who operate mines across B.C.
      </ForwardRef>
      <ForwardRef
        level={4}
      >
        What can I do in MineSpace?
      </ForwardRef>
      <ul
        className="landing-list"
      >
        <li>
          Upload any report specified in the Health, Safety and Reclamation Code
        </li>
        <li>
          View all code variances granted to and incidents reported by your mine
        </li>
        <li>
          View your mine permit and amendment history
        </li>
        <li>
          See all the contacts the Ministry has on file for your mine
        </li>
        <li>
          Find important Ministry contacts
        </li>
      </ul>
      <ForwardRef />
    </Col>
    <Col
      sm={
        Object {
          "span": 24,
        }
      }
      xl={
        Object {
          "span": 12,
        }
      }
      xxl={
        Object {
          "span": 10,
        }
      }
    >
      <ForwardRef
        level={4}
      >
        How do I get access?
      </ForwardRef>
      <ForwardRef>
        You must have a 
        <ForwardRef(Text)
          strong={true}
        >
          Business or Basic BCeID
        </ForwardRef(Text)>
         and then contact us to request access to MineSpace.
      </ForwardRef>
      <ForwardRef
        strong={true}
      >
        If you have a BCeID:
      </ForwardRef>
      <ForwardRef>
        Contact us at 
        <a
          href="mailto:mds@gov.bc.ca"
        >
          mds@gov.bc.ca
        </a>
         to request access to MineSpace.
      </ForwardRef>
      <ForwardRef
        strong={true}
      >
        If you have multiple employees who need to use MineSpace:
      </ForwardRef>
      <ForwardRef>
        <ForwardRef(Text)>
          1. Add them to your Business BCeID
        </ForwardRef(Text)>
        <br />
        <ForwardRef(Text)>
          2. Let us know you want them to be able to access MineSpace
        </ForwardRef(Text)>
      </ForwardRef>
      <ForwardRef
        level={4}
      >
        Don't have a BCeID?
      </ForwardRef>
      <ForwardRef>
        In order to access MineSpace, you need to register for a Business or Basic BCeID. It can take several weeks to process the request, so give yourself plenty of lead time.
      </ForwardRef>
      <ForwardRef>
        Once you have your BCeID, you can add employees and delegates. You can request that anyone added to your Business BCeID account be given access to MineSpace.
      </ForwardRef>
      <Row
        justify="center"
      >
        <Col
          span={24}
        >
          <a
            href="https://www.bceid.ca/register/business/getting_started/getting_started.aspx"
            rel="noopener noreferrer"
            target="_blank"
          >
            <Button
              size="large"
              type="primary"
            >
              Get a BCeID
            </Button>
          </a>
        </Col>
      </Row>
    </Col>
  </Row>
  <Row
    align="top"
    className="landing-section"
    gutter={64}
    justify="center"
  >
    <Col
      sm={
        Object {
          "span": 24,
        }
      }
      xl={
        Object {
          "span": 12,
        }
      }
      xxl={
        Object {
          "span": 10,
        }
      }
    >
      <ForwardRef
        title="Questions?"
      >
        <Row>
          <Col
            span={24}
          >
            <ForwardRef>
              Please let us know about any questions or comments you have regarding your experience using MineSpace.
            </ForwardRef>
            <ForwardRef>
              Email us at 
              <a
                href="mailto:mds@gov.bc.ca"
              >
                mds@gov.bc.ca
              </a>
              .
            </ForwardRef>
          </Col>
        </Row>
      </ForwardRef>
    </Col>
  </Row>
</div>
`;
